lib/msf/core/post/windows/version.rb

```diff
@@ -30,7 +30,13 @@ def get_version_info
       Msf::WindowsVersion.new(major, minor, build, service_pack, product_type)
     else
       build_num_raw = cmd_exec('systeminfo')
-      bn_groups = build_num_raw.match(/OS Version:\s+(\d+)\.(\d+)\.(\d+).*((Service Pack\s+(\d+))|N\/A)/)
+      if build_num_raw.include?('not recognized') # Windows 2K and earlier do not have systeminfo
```

Comment:
Sadly this approach is doomed to fail on non-English locales.

![image](https://user-images.githubusercontent.com/434827/215933106-976b5af7-9260-40d7-86d4-a213a5e04430.png)

A better approach may be to check for known good data, rather than known bad.

Untested suggestion:

```ruby
unless build_num_raw.include?('OS Version:')
  # win 2k
end
```